internal/kube/client.go

```diff
@@ -0,0 +1,368 @@
+// internal/kube/client.go
+// Package kube provides a client-go–backed Kubernetes helper used by the MCP service.
+// It exposes small, deterministic functions for:
+//   • Listing served CRDs
+//   • Validating manifests via server-side dry-run apply
+//   • Fetching a CRD in yaml/json or a short "describe" view
+//   • Preflight connectivity/version check
+package kube
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+	"sync"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	k8syaml "k8s.io/apimachinery/pkg/util/yaml"
+
+	apixclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	apixv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/discovery/cached/memory"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+	"k8s.io/client-go/tools/clientcmd"
+
+	_ "k8s.io/client-go/plugin/pkg/client/auth" // auth providers (GCP, Azure, OIDC, exec)
+	"sigs.k8s.io/yaml"
+)
+
+// Kubectl is a thin wrapper around client-go components used by datumctl.
+// Connections are created lazily on first use.
+type Kubectl struct {
+	// Optional kubeconfig context override.
+	Context string
+
+	// Default namespace used when a namespaced object omits metadata.namespace.
+	Namespace string
+
+	// Reserved for API stability; no effect.
+	Path string
+	Env  []string
+
+	// internal clients (initialized on first use)
+	once    sync.Once
+	initErr error
+
+	cfg    *rest.Config
+	kube   *kubernetes.Clientset
+	apix   *apixclient.Clientset
+	dyn    dynamic.Interface
+	disco  discovery.DiscoveryInterface
+	mapper *restmapper.DeferredDiscoveryRESTMapper
+}
+
+// New returns a new helper; clients are initialized lazily.
+func New() *Kubectl { return &Kubectl{} }
+
+// NewWithRESTConfig returns a helper that uses the provided *rest.Config.
+// When set, kubeconfig will NOT be loaded; clients will be built from this cfg.
+func NewWithRESTConfig(cfg *rest.Config) *Kubectl { return &Kubectl{cfg: cfg} }
+
+// CRDItem is a compact view of a CustomResourceDefinition.
+type CRDItem struct {
+	Name     string   `json:"name"`     // e.g. "httpproxies.networking.datumapis.com"
+	Group    string   `json:"group"`    // e.g. "networking.datumapis.com"
+	Kind     string   `json:"kind"`     // Kind from CRD spec
+	Versions []string `json:"versions"` // served versions
+}
```

Comment:
This might be better placed under the `mcp` package since it's really just the view layer the MCP server wants to render to users when listing CRDs. I'd expect any client to expose the ordinary ListCustomResourceDefinitions endpoint that's available on the API.